tests/main.nix

```diff
@@ -0,0 +1,69 @@
+# Collects the various test modules in tests/test-sources/ and groups them into a number of test derivations
```

Comment:
Is there a better name than `main.nix`?

- `main.nix` could be confusing, as `default.nix` is the "main" entrypoint to a directory
- `modules.nix` could make sense (the file is responsible for making tests from the modules in `test-sources`), but we already have a `modules.nix` that tests the wrapper modules...
- `generated.nix` could make sence too, but we already have a `generated.nix` test that tests the _ci-generated files_ are valid

Maybe `modules.nix` is the way to go, if we renamed the current `modules.nix` to `wrappers.nix`? But this is making the diff much larger than it really should be for a minor refactoring PR...

Reply:
I agree `modules.nix` would be better. We can do it in another PR though.